### PR TITLE
Move to PHP8.0 syntax in Components

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

We can finally do that 🚀 ❤️

### CODE REVIEW

**Comment by Roshyo** [comment]
Do you really think that promoted properties improve readability ?

File: src/Sylius/Component/Core/OrderProcessing/OrderTaxesProcessor.php [hunk 1]
```diff
@@ -31,25 +31,12 @@
...
-
     public function __construct(
-        ZoneProviderInterface $defaultTaxZoneProvider,
-        ZoneMatcherInterface $zoneMatcher,
-        PrioritizedServiceRegistryInterface $strategyRegistry,
-        ?TaxationAddressResolverInterface $taxationAddressResolver = null
+        private ZoneProviderInterface $defaultTaxZoneProvider,
+        private ZoneMatcherInterface $zoneMatcher,
+        private PrioritizedServiceRegistryInterface $strategyRegistry,
+        private ?\Sylius\Component\Core\Resolver\TaxationAddressResolverInterface $taxationAddressResolver = null
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/OrderProcessing/OrderTaxesProcessor.php:38]:
```suggestion
        private ?TaxationAddressResolverInterface $taxationAddressResolver = null
```

File: src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php [hunk 2]
```diff
@@ -24,16 +24,8 @@
...
-
-    private string $targetState;
-
-    public function __construct(
-        OrderPaymentProviderInterface $orderPaymentProvider,
-        string $targetState = PaymentInterface::STATE_CART
-    ) {
-        $this->orderPaymentProvider = $orderPaymentProvider;
-        $this->targetState = $targetState;
+    public function __construct(private OrderPaymentProviderInterface $orderPaymentProvider, private string $targetState = PaymentInterface::STATE_CART)
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php:27]: should be split into multiple lines

**Review by lchrusciel — APPROVED** [review]
I'm in favour. I would say, that readability may be subjective and for sure, I'm not yet used to the new syntax. But for sure it is one place less when someone may do mistake

File: src/Sylius/Component/Core/Taxation/Applicator/OrderItemUnitsTaxesApplicator.php [hunk 3]
```diff
@@ -24,20 +24,8 @@
...
-    public function __construct(
-        CalculatorInterface $calculator,
-        AdjustmentFactoryInterface $adjustmentFactory,
-        TaxRateResolverInterface $taxRateResolver
-    ) {
-        $this->calculator = $calculator;
-        $this->adjustmentFactory = $adjustmentFactory;
-        $this->taxRateResolver = $taxRateResolver;
+    public function __construct(private CalculatorInterface $calculator, private AdjustmentFactoryInterface $adjustmentFactory, private TaxRateResolverInterface $taxRateResolver)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Taxation/Applicator/OrderItemUnitsTaxesApplicator.php:28]:
```suggestion
    ```suggestion
    public function __construct(
        private CalculatorInterface $calculator,
        private AdjustmentFactoryInterface $adjustmentFactory,
        private TaxRateResolverInterface $taxRateResolver
    ) {
```
```

File: src/Sylius/Component/Core/Context/ShopperContext.php [hunk 4]
```diff
@@ -25,24 +25,8 @@
...
-        CurrencyContextInterface $currencyContext,
-        LocaleContextInterface $localeContext,
-        CustomerContextInterface $customerContext
-    ) {
-        $this->channelContext = $channelContext;
-        $this->currencyContext = $currencyContext;
-        $this->localeContext = $localeContext;
-        $this->customerContext = $customerContext;
+    public function __construct(private ChannelContextInterface $channelContext, private CurrencyContextInterface $currencyContext, private LocaleContextInterface $localeContext, private CustomerContextInterface $customerContext)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Context/ShopperContext.php:29]:
```suggestion
    ```suggestion
    public function __construct(
        private ChannelContextInterface $channelContext,
        private CurrencyContextInterface $currencyContext,
        private LocaleContextInterface $localeContext,
        private CustomerContextInterface $customerContext
    ) {
```
```

File: src/Sylius/Component/Core/Dashboard/SalesSummary.php [hunk 5]
```diff
@@ -18,13 +18,11 @@
...
-    private array $intervalsSalesMap = [];
-
     public function __construct(
-        array $salesData
-    ) {
-        $this->intervalsSalesMap = $salesData;
+        /** @psalm-var array<string, string> */
+        private array $intervalsSalesMap
+    )
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Dashboard/SalesSummary.php:25]:
```suggestion
    ) {
```

File: src/Sylius/Component/Core/Locale/Context/StorageBasedLocaleContext.php [hunk 6]
```diff
@@ -22,20 +22,8 @@
...
-    public function __construct(
-        ChannelContextInterface $channelContext,
-        LocaleStorageInterface $localeStorage,
-        LocaleProviderInterface $localeProvider
-    ) {
-        $this->channelContext = $channelContext;
-        $this->localeStorage = $localeStorage;
-        $this->localeProvider = $localeProvider;
+    public function __construct(private ChannelContextInterface $channelContext, private LocaleStorageInterface $localeStorage, private LocaleProviderInterface $localeProvider)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Locale/Context/StorageBasedLocaleContext.php:26]:
```suggestion
    
    public function __construct(
        private ChannelContextInterface $channelContext,
        private LocaleStorageInterface $localeStorage,
        private LocaleProviderInterface $localeProvider
    ) {
```

File: src/Sylius/Component/Core/Payment/Provider/OrderPaymentProvider.php [hunk 7]
```diff
@@ -26,20 +26,8 @@
...
-    public function __construct(
-        DefaultPaymentMethodResolverInterface $defaultPaymentMethodResolver,
-        PaymentFactoryInterface $paymentFactory,
-        StateMachineFactoryInterface $stateMachineFactory
-    ) {
-        $this->defaultPaymentMethodResolver = $defaultPaymentMethodResolver;
-        $this->paymentFactory = $paymentFactory;
-        $this->stateMachineFactory = $stateMachineFactory;
+    public function __construct(private DefaultPaymentMethodResolverInterface $defaultPaymentMethodResolver, private PaymentFactoryInterface $paymentFactory, private StateMachineFactoryInterface $stateMachineFactory)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Payment/Provider/OrderPaymentProvider.php:30]:
```suggestion
    public function __construct(
        private DefaultPaymentMethodResolverInterface $defaultPaymentMethodResolver,
        private PaymentFactoryInterface $paymentFactory,
        private StateMachineFactoryInterface $stateMachineFactory
    ) {
```

File: src/Sylius/Component/Core/Resolver/EligibleDefaultShippingMethodResolver.php [hunk 8]
```diff
@@ -29,20 +29,8 @@
...
-    public function __construct(
-        ShippingMethodRepositoryInterface $shippingMethodRepository,
-        ShippingMethodEligibilityCheckerInterface $shippingMethodEligibilityChecker,
-        ZoneMatcherInterface $zoneMatcher
-    ) {
-        $this->shippingMethodRepository = $shippingMethodRepository;
-        $this->shippingMethodEligibilityChecker = $shippingMethodEligibilityChecker;
-        $this->zoneMatcher = $zoneMatcher;
+    public function __construct(private ShippingMethodRepositoryInterface $shippingMethodRepository, private ShippingMethodEligibilityCheckerInterface $shippingMethodEligibilityChecker, private ZoneMatcherInterface $zoneMatcher)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Resolver/EligibleDefaultShippingMethodResolver.php:33]:
```suggestion
    public function __construct(
        private ShippingMethodRepositoryInterface $shippingMethodRepository, 
        private ShippingMethodEligibilityCheckerInterface $shippingMethodEligibilityChecker, 
        private ZoneMatcherInterface $zoneMatcher
    ) {
```

File: src/Sylius/Component/Core/Resolver/ZoneAndChannelBasedShippingMethodsResolver.php [hunk 9]
```diff
@@ -25,20 +25,8 @@
...
-    public function __construct(
-        ShippingMethodRepositoryInterface $shippingMethodRepository,
-        ZoneMatcherInterface $zoneMatcher,
-        ShippingMethodEligibilityCheckerInterface $eligibilityChecker
-    ) {
-        $this->shippingMethodRepository = $shippingMethodRepository;
-        $this->zoneMatcher = $zoneMatcher;
-        $this->eligibilityChecker = $eligibilityChecker;
+    public function __construct(private ShippingMethodRepositoryInterface $shippingMethodRepository, private ZoneMatcherInterface $zoneMatcher, private ShippingMethodEligibilityCheckerInterface $eligibilityChecker)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Resolver/ZoneAndChannelBasedShippingMethodsResolver.php:29]:
```suggestion
    
    public function __construct(
        private ShippingMethodRepositoryInterface $shippingMethodRepository,
        private ZoneMatcherInterface $zoneMatcher,
        private ShippingMethodEligibilityCheckerInterface $eligibilityChecker
    ) {
```

File: src/Sylius/Component/Core/StateResolver/CheckoutStateResolver.php [hunk 10]
```diff
@@ -22,20 +22,8 @@
...
-    public function __construct(
-        FactoryInterface $stateMachineFactory,
-        OrderPaymentMethodSelectionRequirementCheckerInterface $orderPaymentMethodSelectionRequirementChecker,
-        OrderShippingMethodSelectionRequirementCheckerInterface $orderShippingMethodSelectionRequirementChecker
-    ) {
-        $this->stateMachineFactory = $stateMachineFactory;
-        $this->orderPaymentMethodSelectionRequirementChecker = $orderPaymentMethodSelectionRequirementChecker;
-        $this->orderShippingMethodSelectionRequirementChecker = $orderShippingMethodSelectionRequirementChecker;
+    public function __construct(private FactoryInterface $stateMachineFactory, private OrderPaymentMethodSelectionRequirementCheckerInterface $orderPaymentMethodSelectionRequirementChecker, private OrderShippingMethodSelectionRequirementCheckerInterface $orderShippingMethodSelectionRequirementChecker)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/StateResolver/CheckoutStateResolver.php:26]:
```suggestion
    public function __construct(
        private FactoryInterface $stateMachineFactory,
        private OrderPaymentMethodSelectionRequirementCheckerInterface $orderPaymentMethodSelectionRequirementChecker, 
        private OrderShippingMethodSelectionRequirementCheckerInterface $orderShippingMethodSelectionRequirementChecker
    ) {
```

File: src/Sylius/Component/Core/Taxation/Applicator/OrderItemsTaxesApplicator.php [hunk 11]
```diff
@@ -26,24 +26,8 @@
...
-        AdjustmentFactoryInterface $adjustmentFactory,
-        IntegerDistributorInterface $distributor,
-        TaxRateResolverInterface $taxRateResolver
-    ) {
-        $this->calculator = $calculator;
-        $this->adjustmentFactory = $adjustmentFactory;
-        $this->distributor = $distributor;
-        $this->taxRateResolver = $taxRateResolver;
+    public function __construct(private CalculatorInterface $calculator, private AdjustmentFactoryInterface $adjustmentFactory, private IntegerDistributorInterface $distributor, private TaxRateResolverInterface $taxRateResolver)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Taxation/Applicator/OrderItemsTaxesApplicator.php:30]:
```suggestion
    public function __construct(
        private CalculatorInterface $calculator,
        private AdjustmentFactoryInterface $adjustmentFactory,
        private IntegerDistributorInterface $distributor,
        private TaxRateResolverInterface $taxRateResolver
    ) {
```

File: src/Sylius/Component/Core/Taxation/Applicator/OrderShipmentTaxesApplicator.php [hunk 12]
```diff
@@ -26,20 +26,8 @@
...
-    public function __construct(
-        CalculatorInterface $calculator,
-        AdjustmentFactoryInterface $adjustmentFactory,
-        TaxRateResolverInterface $taxRateResolver
-    ) {
-        $this->calculator = $calculator;
-        $this->adjustmentFactory = $adjustmentFactory;
-        $this->taxRateResolver = $taxRateResolver;
+    public function __construct(private CalculatorInterface $calculator, private AdjustmentFactoryInterface $adjustmentFactory, private TaxRateResolverInterface $taxRateResolver)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Taxation/Applicator/OrderShipmentTaxesApplicator.php:30]:
```suggestion
    public function __construct(
        private CalculatorInterface $calculator,
        private AdjustmentFactoryInterface $adjustmentFactory,
        private TaxRateResolverInterface $taxRateResolver
    ) {
```

File: src/Sylius/Component/Core/Test/Services/DefaultChannelFactory.php [hunk 13]
```diff
@@ -28,36 +28,8 @@ final class DefaultChannelFactory implements DefaultChannelFactoryInterface
...
-    ) {
-        $this->channelFactory = $channelFactory;
-        $this->currencyFactory = $currencyFactory;
-        $this->localeFactory = $localeFactory;
-        $this->channelRepository = $channelRepository;
-        $this->currencyRepository = $currencyRepository;
-        $this->localeRepository = $localeRepository;
-        $this->defaultLocaleCode = $defaultLocaleCode;
+    public function __construct(private ChannelFactoryInterface $channelFactory, private FactoryInterface $currencyFactory, private FactoryInterface $localeFactory, private RepositoryInterface $channelRepository, private RepositoryInterface $currencyRepository, private RepositoryInterface $localeRepository, private string $defaultLocaleCode)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Test/Services/DefaultChannelFactory.php:32]:
```suggestion
    public function __construct(
        private ChannelFactoryInterface $channelFactory,
        private FactoryInterface $currencyFactory, 
        private FactoryInterface $localeFactory, 
        private RepositoryInterface $channelRepository, 
        private RepositoryInterface $currencyRepository, 
        private RepositoryInterface $localeRepository, 
        private string $defaultLocaleCode
    ) {
```

File: src/Sylius/Component/Core/Test/Services/DefaultUnitedStatesChannelFactory.php [hunk 14]
```diff
@@ -37,52 +37,8 @@ final class DefaultUnitedStatesChannelFactory implements DefaultChannelFactoryIn
...
-        $this->localeRepository = $localeRepository;
-        $this->zoneRepository = $zoneRepository;
-        $this->channelFactory = $channelFactory;
-        $this->countryFactory = $countryFactory;
-        $this->currencyFactory = $currencyFactory;
-        $this->localeFactory = $localeFactory;
-        $this->zoneFactory = $zoneFactory;
-        $this->defaultLocaleCode = $defaultLocaleCode;
+    public function __construct(private RepositoryInterface $channelRepository, private RepositoryInterface $countryRepository, private RepositoryInterface $currencyRepository, private RepositoryInterface $localeRepository, private RepositoryInterface $zoneRepository, private ChannelFactoryInterface $channelFactory, private FactoryInterface $countryFactory, private FactoryInterface $currencyFactory, private FactoryInterface $localeFactory, private ZoneFactoryInterface $zoneFactory, private string $defaultLocaleCode)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Test/Services/DefaultUnitedStatesChannelFactory.php:41]:
```suggestion
    public function __construct(
        private RepositoryInterface $channelRepository, 
        private RepositoryInterface $countryRepository, 
        private RepositoryInterface $currencyRepository, 
        private RepositoryInterface $localeRepository, 
        private RepositoryInterface $zoneRepository, 
        private ChannelFactoryInterface $channelFactory, 
        private FactoryInterface $countryFactory, 
        private FactoryInterface $currencyFactory, 
        private FactoryInterface $localeFactory, 
        private ZoneFactoryInterface $zoneFactory, 
        private string $defaultLocaleCode
    ) {
```

File: src/Sylius/Component/Order/Modifier/OrderModifier.php [hunk 15]
```diff
@@ -19,16 +19,8 @@
...
-    private OrderItemQuantityModifierInterface $orderItemQuantityModifier;
-
-    public function __construct(
-        OrderProcessorInterface $orderProcessor,
-        OrderItemQuantityModifierInterface $orderItemQuantityModifier
-    ) {
-        $this->orderProcessor = $orderProcessor;
-        $this->orderItemQuantityModifier = $orderItemQuantityModifier;
+    public function __construct(private OrderProcessorInterface $orderProcessor, private OrderItemQuantityModifierInterface $orderItemQuantityModifier)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Order/Modifier/OrderModifier.php:23]:
```suggestion
    public function __construct(
        private OrderProcessorInterface $orderProcessor,
        private OrderItemQuantityModifierInterface $orderItemQuantityModifier
    ) {
```

File: src/Sylius/Component/Promotion/Generator/PromotionCouponGenerator.php [hunk 16]
```diff
@@ -23,24 +23,8 @@
...
-        PromotionCouponRepositoryInterface $couponRepository,
-        ObjectManager $objectManager,
-        GenerationPolicyInterface $generationPolicy
-    ) {
-        $this->couponFactory = $couponFactory;
-        $this->couponRepository = $couponRepository;
-        $this->objectManager = $objectManager;
-        $this->generationPolicy = $generationPolicy;
+    public function __construct(private FactoryInterface $couponFactory, private PromotionCouponRepositoryInterface $couponRepository, private ObjectManager $objectManager, private GenerationPolicyInterface $generationPolicy)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Promotion/Generator/PromotionCouponGenerator.php:27]:
```suggestion
    public function __construct(
        private FactoryInterface $couponFactory, 
        private PromotionCouponRepositoryInterface $couponRepository, 
        private ObjectManager $objectManager, 
        private GenerationPolicyInterface $generationPolicy
    ) {
```

File: src/Sylius/Component/Promotion/Processor/PromotionProcessor.php [hunk 17]
```diff
@@ -20,20 +20,8 @@
...
-    public function __construct(
-        PreQualifiedPromotionsProviderInterface $preQualifiedPromotionsProvider,
-        PromotionEligibilityCheckerInterface $promotionEligibilityChecker,
-        PromotionApplicatorInterface $promotionApplicator
-    ) {
-        $this->preQualifiedPromotionsProvider = $preQualifiedPromotionsProvider;
-        $this->promotionEligibilityChecker = $promotionEligibilityChecker;
-        $this->promotionApplicator = $promotionApplicator;
+    public function __construct(private PreQualifiedPromotionsProviderInterface $preQualifiedPromotionsProvider, private PromotionEligibilityCheckerInterface $promotionEligibilityChecker, private PromotionApplicatorInterface $promotionApplicator)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Promotion/Processor/PromotionProcessor.php:24]:
```suggestion
    public function __construct(
        private PreQualifiedPromotionsProviderInterface $preQualifiedPromotionsProvider,
        private PromotionEligibilityCheckerInterface $promotionEligibilityChecker,
        private PromotionApplicatorInterface $promotionApplicator
    )
    {
```

File: src/Sylius/Component/Shipping/Resolver/ShippingMethodsResolver.php [hunk 18]
```diff
@@ -19,16 +19,8 @@
...
-    private ShippingMethodEligibilityCheckerInterface $eligibilityChecker;
-
-    public function __construct(
-        ObjectRepository $shippingMethodRepository,
-        ShippingMethodEligibilityCheckerInterface $eligibilityChecker
-    ) {
-        $this->shippingMethodRepository = $shippingMethodRepository;
-        $this->eligibilityChecker = $eligibilityChecker;
+    public function __construct(private ObjectRepository $shippingMethodRepository, private ShippingMethodEligibilityCheckerInterface $eligibilityChecker)
+    {
```
Comment (lchrusciel) [inline src/Sylius/Component/Shipping/Resolver/ShippingMethodsResolver.php:23]:
```suggestion
    public function __construct(
        private ObjectRepository $shippingMethodRepository,
        private ShippingMethodEligibilityCheckerInterface $eligibilityChecker
    ) {
```

**Review by lchrusciel — COMMENTED** [review]
a few more places

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Mateusz! :tada:

File: rector.php [hunk 19]
```diff
@@ -10,6 +10,8 @@
 
 return static function (ContainerConfigurator $containerConfigurator): void
 {
+    $containerConfigurator->import(SetList::PHP_80);
```
Comment (TomasVotruba) [inline rector.php:13]: Just a tip for esaier maintenance of PHP version. There is now "up to" one liner: https://getrector.org/blog/new-in-rector-012-the-latest-php-in-a-single-import

**Comment by TomasVotruba** [comment]
Great job with upgrade :+1: 

I love the diff :laughing: 

image

**Pull request merged**